internal/builder/docker.go

```diff
@@ -0,0 +1,130 @@
+package builder
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"runtime"
+
+	"github.com/lucor/fyne-cross/internal/volume"
+)
+
+const (
+	// DefaultDockerImage is the remote docker image
+	// name used by default
+	DefaultDockerImage = "lucor/fyne-cross:1.4.0"
+)
+
+type dockerImageProvider interface {
```

Comment:
What about having only the Builder responsible for the definition of the custom docker image ?
Adding something like `DockerImage() string` and `SetDockerImage(string)` could be good enough?